### ISSUE TITLE
vue/valid-v-model bug on v-for variable destructuring

### ISSUE THREAD

**Issue by serenedu** [vuejs/eslint-plugin-vue#900]
**Tell us about your environment**	
* **ESLint version:** 5.16.0
* **eslint-plugin-vue version:** 5.2.2	
* **Node version:** 10.15.3

**Please show your full configuration:**

```js
module.exports = {
  root: true,
  env: {
    node: true,
  },
  extends: ['plugin:vue/essential', '@vue/prettier'],
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
  },
  parserOptions: {
    parser: 'babel-eslint',
  },
}
```

**What did you do?**	

```vue
<div v-for="{ name, id } in array" :key="id">
    <input type="text" v-model="name" />
</div>
```

**What did you expect to happen?**
no error

**What actually happened?**

eslint throws error`'v-model' directives cannot update the iteration variable 'name' itself.`
Although this usage is valid and doesn't output error in the browser console.
